avalon/mongodb.py

```diff
@@ -390,17 +390,26 @@ def projects(self, query_filter=None, projection=None):
 
         find_args = [query_filter]
         if projection:
-            find_args.append(projection)
+            _projection = projection.copy()
+            _projection.update({"data.active": 1})
+            find_args.append(_projection)
 
         for project_name in self._database.collection_names():
             if project_name in ("system.indexes",):
                 continue
 
             # Each collection will have exactly one project document
 
-            document = self._database[project_name].find_one(*find_args)
-            if document is not None:
-                yield document
+            doc = self._database[project_name].find_one(*find_args)
+            if doc is not None:
+
+                if only_active and not doc.get("data", {}).get("active", True):
+                    continue
+
+                if projection and not projection.get("data.active"):
+                    doc.get("data", {}).pop("active", None)
```

Comment:
It is possible that only `"data"` were specified in projection.
```suggestion
                if (
                    projection
                    and "data" not in projection
                    and "data.active" not in projection
                ):
                    doc.get("data", {}).pop("active", None)
```

It is a question if the key should be popped out?

Reply:
Ah, hmmm...
Maybe instead of altering the projection, adding `$or: [{$exists: {'data.active': 0}}, {'data.active': True}]` to query is better ?

Reply:
Query is for finding documents but projection says which data will be returned...

Reply:
Yeah, but all we need to filter the deactivated project is to check the `data.active`, so if the project document not matching the query, we could skip it and no need to project. Right ?

Reply:
Yes, you're right...

Reply:
So it could be:
```
query_filter = copy.deepcopy(query_filter)
if only_active:
    query_filter["$or"] = [{"$exists": {"data.active": 0}}, {"data.active": True}]
```

And keep projection unchanged. Goood

Reply:
But there can be already `"$or"` in the query...

Reply:
In the end I think that just keeping the active key in result is good enough.

Reply:
Was wondering if we could deprecate the `query_filter` arg ? Looks like it's not being used elsewhere. Or ignore the `only_active` if the `query_filter` is not None ?

Reply:
True `query_filter` is not used and it's purpose was to be able query active so we can get rid of it...